node/src/components/storage.rs

```diff
@@ -989,3 +1019,48 @@ impl Storage {
         switch_block
     }
 }
+
+/// Utility function to check the integrity of the block_body database at bringup.
+fn check_block_body_db(env: &Environment, block_body_db: &Database) -> Result<(), LmdbExtError> {
+    info!("Checking block body db");
+    let txn = env.begin_ro_txn()?;
+    let mut cursor = txn.open_ro_cursor(*block_body_db)?;
+
+    for (raw_key, raw_val) in cursor.iter() {
+        let body: BlockBody = lmdb_ext::deserialize(raw_val)?;
+        assert_eq!(
+            raw_key,
+            body.hash().as_ref(),
+            "found corrupt block body in database"
+        );
+    }
+    info!("block body db check complete");
+    Ok(())
+}
+
+/// Utility function to check the integrity of the block_metadata database at bringup.
+fn check_block_metadata_db(
+    env: &Environment,
+    block_metadata_db: &Database,
+) -> Result<(), LmdbExtError> {
+    info!("Checking block_metadata_db");
+    let txn = env.begin_ro_txn()?;
+    let mut cursor = txn.open_ro_cursor(*block_metadata_db)?;
+
+    for (raw_key, raw_val) in cursor.iter() {
+        let signature: BlockSignatures = lmdb_ext::deserialize(raw_val)?;
+        // Signature verification could be very slow process
+        // It iterates over every signature and verifies them.
+        match signature.verify() {
+            Ok(_) => continue,
```

Comment:
We can't `continue` here, since then we won't go to the `assert_eq!` below.

Reply:
Moving the `assert_eq!` into the match statement

Reply:
Removed `continue` and moved `assert_eq` into the match statement